### DISCUSSION TITLE
How to ignore UUID type when using it with .or()?

### DISCUSSION THREAD

**Discussion by z-x** [supabase/supabase#12725] [Questions] [answered]
Hello!
So I am passing a param in the URL that is either the user's nickname or his UUID. So it can be either `UUID` or `text`. Then, based on that I would like to use the `or()` method in JavaScript to select the row if it matches either UUID or the `username`:

```js
.or(`username.eq.${params.username},id.eq.${params.username}`)
```

The above works if I pass the UUID as `params.username`, but fails when I pass a text, because it expects the `id` column to be UUID. Is there a way to tell supabase to just keep it calm and try get me the results even if the second argument in `or` statement is wrong type?

I could, in theory, check if the param matches the UUID format with regexp, but am wondering if there might be a simpler way?

**Comment by GaryAustin1** [comment, marked as answer]
This was answered in Discord.
There is no way to ignore the column or parameter type not matching AND in Postgres there is no guarantee which or condition is run first.
The best way is to use conditional chaining of the .eq() based on which is valid.
